dinstaller-lib/share/profile.schema.json

```diff
@@ -0,0 +1,109 @@
+{
+  "$schema": "https://json-schema.org/draft-07/schema",
+  "$id": "https://github.com/d-installer/profile.schema.json",
+  "title": "Profile",
+  "description": "Profile definition for automated installation",
+  "type": "object",
+  "additionalProperties": false,
+  "properties": {
+    "software": {
+      "description": "Software settings",
+      "type": "object",
+      "properties": {
+        "product": {
+          "description": "Product identifier",
+          "type": "string"
+        }
+      },
+      "required": [
+        "product"
+      ]
+    },
+    "user": {
+      "description": "User settings",
+      "type": "object",
+      "properties": {
+        "fullName": {
+          "description": "User full name",
+          "type": "string"
+        },
+        "userName": {
+          "description": "User login name",
+          "type": "string"
+        },
+        "password": {
+          "description": "User password",
+          "type": "string"
+        }
+      },
+      "required": [
```

Comment:
same here. How hard is to express something like root password or root ssh pub key or user?

Reply:
Good point. I will give it a try.

Reply:
I will update the schema when I add support for root authentication, in a separate PR.